server/coordinator/queue/delay_queue.go

```diff
@@ -0,0 +1,128 @@
+// Copyright 2023 CeresDB Project Authors. Licensed under Apache-2.0.
+
+package queue
+
+import (
+	"container/heap"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/CeresDB/ceresmeta/server/coordinator/procedure"
+	"github.com/pkg/errors"
+)
+
+type procedureScheduleEntry struct {
+	procedure procedure.Procedure
+	runAfter  time.Time
+}
+
+type ProcedureDelayQueue struct {
+	maxLen int
+
+	// This lock is used to protect the following fields.
+	lock      sync.RWMutex
+	heapQueue *heapPriorityQueue
+	// existingProcs is used to record procedures has been pushed into the queue,
+	// and they will be used to verify the addition of duplicate elements.
+	existingProcs map[uint64]struct{}
+}
+
+// heapPriorityQueue is no internal lock,
+// and its thread safety is guaranteed by the external caller.
+type heapPriorityQueue struct {
+	procedures []*procedureScheduleEntry
+}
+
+func (q *heapPriorityQueue) Len() int {
+	return len(q.procedures)
+}
+
+// The dequeue order of elements is determined by the less method.
+// When return procedures[i].runAfter < procedures[j].runAfter, the element with smallest will be pop first.
+func (q *heapPriorityQueue) Less(i, j int) bool {
+	return q.procedures[i].runAfter.Before(q.procedures[j].runAfter)
```

Comment:
In order to make the smallest element located at the last in the slice, the compare must be negative, that is to say:
```
```suggestion
	return q.procedures[j].runAfter.Before(q.procedures[i].runAfter)
```
```

Reply:
This is my fault. This PR is ok now.